File: backend/infraction-service/app/com/radovan/play/services/PrometheusService.scala
package com.radovan.play.services

trait PrometheusService {

  def increaseRequestCount(): Unit

  def recordResponseTime(duration: Double): Unit

  def updateMemoryUsage(): Unit

  def updateThreadCount(): Unit

  def updateCpuLoad(): Unit

  def updateDatabaseQueryCount(): Unit

  def updateHeapAllocationRate(): Unit

  def updateActiveSessions(): Unit

  def updateHttpStatusCount(statusCode: Int): Unit

  def updateExternalApiLatency(duration: Double): Unit


}

File: backend/radar-service/src/main/java/com/radovan/spring/services/PrometheusService.java
package com.radovan.spring.services;


public interface PrometheusService {

    void increaseRequestCount();

    void recordResponseTime(double duration);

    void updateMemoryUsage();

    void updateThreadCount();

    void updateCpuLoad();

    void updateDatabaseQueryCount();

    void updateHeapAllocationRate();

    void updateActiveSessions();

    void updateHttpStatusCount(int statusCode);

    void updateExternalApiLatency(double duration);

    void updateControllerMethodTags(String controller, String method);
}
